### PR TITLE
curvefs/metaserver: tuning rocksdb for reduce range scan performance …

### PR DESCRIPTION
**Pull request by Wine93**
…skews caused by scanning excessed number of tombstones (#1679).

Signed-off-by: Wine93 <wine93.info@gmail.com>

### What problem does this PR solve?

Issue Number: #1679

Problem Summary: range scan performance skews caused by scanning excessed number of tombstones

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: curvefs/src/metaserver/storage/rocksdb_options.cpp [hunk 1]
```diff
@@ -158,12 +179,18 @@ void InitRocksdbOptions(
...
+    defaultCfOptions.max_bytes_for_level_base =
+        FLAGS_rocksdb_max_bytes_for_level_base;
     defaultCfOptions.prefix_extractor.reset(
         rocksdb::NewFixedPrefixTransform(RocksDBStorage::GetKeyPrefixLength()));
     defaultCfOptions.memtable_prefix_bloom_size_ratio =
         FLAGS_rocksdb_memtable_prefix_bloom_size_ratio;
     defaultCfOptions.table_factory.reset(
         rocksdb::NewBlockBasedTableFactory(tableOptions));
+    defaultCfOptions.table_properties_collector_factories.emplace_back(
+        rocksdb::NewCompactOnDeletionCollectorFactory(10000, 1000));
```
Comment (wu-hanqing) [inline curvefs/src/metaserver/storage/rocksdb_options.cpp:193]: I think the thrid argument `deletion_ratio` might be also usefull. And, please don't hard code 10'000 1'000 in here, give them more meaningfull names.

File: curvefs/src/metaserver/storage/rocksdb_storage.h [hunk 2]
```diff
@@ -35,8 +35,10 @@
 #include "rocksdb/options.h"
 #include "rocksdb/filter_policy.h"
 #include "rocksdb/slice_transform.h"
+#include "rocksdb/table_properties.h"
```
Comment (wu-hanqing) [inline curvefs/src/metaserver/storage/rocksdb_storage.h:38]: it seems you didn't directly use names from this header file in current header file, so put these include in source file in enough.

**Review by SeanHai — APPROVED** [review]

File: curvefs/src/metaserver/storage/rocksdb_storage.h (hunk 2 above)
Comment (Wine93) [reply curvefs/src/metaserver/storage/rocksdb_storage.h:38]: > it seems you didn't directly use names from this header file in current header file, so put these include in source file in enough.

fixed.

File: curvefs/src/metaserver/storage/rocksdb_options.cpp (hunk 1 above)
Comment (Wine93) [reply curvefs/src/metaserver/storage/rocksdb_options.cpp:193]: > I think the thrid argument `deletion_ratio` might be also usefull. And, please don't hard code 10'000 1'000 in here, give them more meaningfull names.

fixed.

**Review by wu-hanqing — APPROVED** [review]

**Pull request merged**